src/edg.rs

```diff
@@ -14,26 +16,34 @@ pub trait ExtendedDependencyGraph<V: Hash + Eq + PartialEq + Clone> {
 }
 
 pub fn distributed_certain_zero<
-    G: ExtendedDependencyGraph<V> + Send + Sync + Clone + 'static,
-    V: Hash + Eq + PartialEq + Clone + Send + Sync + 'static,
+    G: ExtendedDependencyGraph<V> + Send + Sync + Clone + Debug + 'static,
+    V: Hash + Eq + PartialEq + Clone + Send + Sync + Debug + 'static,
 >(
     edg: G,
     v0: V,
     worker_count: u64,
-) {
+) -> VertexAssignment {
     // NOTE: 'static lifetime doesn't mean the full duration of the program execution
     let (broker, mut msg_rxs, mut term_rxs) = ChannelBroker::new(worker_count);
     let broker = Arc::new(broker);
+    let (mut tx, mut rx) = crossbeam_channel::bounded(worker_count as usize);
 
     for i in (0..worker_count).rev() {
         let msg_rx = msg_rxs.pop().unwrap();
         let term_rx = term_rxs.pop().unwrap();
         let mut worker = Worker::new(i, worker_count, v0.clone(), msg_rx, term_rx, broker.clone(), edg.clone());
-        thread::spawn(move || worker.run());
+        let tx = tx.clone();
+        thread::spawn(move || {
+            let result = worker.run();
+            tx.send(result).expect("Failed to submit final assignment of v0");
+        });
     }
+
```

Comment:
Kommer alle workers til at kende final assignment af v0?

Reply:
Dårlig markering. Jeg refererer til dette:
```rust
    for i in (0..worker_count).rev() {
        let msg_rx = msg_rxs.pop().unwrap();
        let msg_rx = msg_rxs.pop().unwrap();
        let term_rx = term_rxs.pop().unwrap();
        let term_rx = term_rxs.pop().unwrap();
        let mut worker = Worker::new(i, worker_count, v0.clone(), msg_rx, term_rx, broker.clone(), edg.clone());
        let mut worker = Worker::new(i, worker_count, v0.clone(), msg_rx, term_rx, broker.clone(), edg.clone());
        thread::spawn(move || worker.run());
        let tx = tx.clone();
        thread::spawn(move || {
            let result = worker.run();
            tx.send(result).expect("Failed to submit final assignment of v0");
        });
    }
    rx.recv().unwrap()
```

Reply:
Min forståelse af algorithme 1 fra _Extended Dependency Graphs and Efficient Distributed Fixed-Point Computation_ så kommer alle workers til et svar, og da der ikke er nævnt noget om hvilken worker man skal tage sit svar fra så antager jeg de alle kommer til det samme svar

Reply:
Jeg har kigget lidt dybere i koden og kan se, at assignment værdien af v0 er sent som en del af termineringsbeskeden, så jeg tror virker som det er 👍